### PR TITLE
fix: fix kafka wal logs deletion and recovery logic

### PR DESCRIPTION
**Pull request by Rachelint**
## Rationale
In the old kafka logs deletion and recovery logic, some conercases leading to logs can never be cleaned up.

## Detailed Changes
+ Fix the logs deletion logic.
+ Fix recovery logic.

## Test Plan
Test manually.

### CODE REVIEW

File: wal/src/kv_encoder.rs [hunk 1]
```diff
@@ -115,17 +115,15 @@ pub enum Namespace {
 }
 
 /// Log key in old wal design, map the `TableId` to `RegionId`
-#[allow(unused)]
+
```
Comment (chunshao90) [inline wal/src/kv_encoder.rs]: Remove blank line.

File: wal/src/kv_encoder.rs [hunk 2]
```diff
@@ -589,7 +581,7 @@ impl LogBatchEncoder {
 }
 
 /// Common log key used in multiple wal implementation
-#[allow(unused)]
+
```
Comment (chunshao90) [inline wal/src/kv_encoder.rs]: Remove blank line.

File: wal/src/message_queue_impl/namespace.rs [hunk 3]
```diff
@@ -31,7 +31,7 @@ use crate::{
 #[derive(Debug, Snafu)]
 pub enum Error {
     #[snafu(display(
-        "Failed to open region, namespace:{}, location:{:?}, err:{}",
+        "Failed to get sequence, namespace:{}, location:{:?}, err:{}",
```
Comment (chunshao90) [inline wal/src/message_queue_impl/namespace.rs:34]:
```suggestion
        "Failed to get sequence, namespace:{namespace}, location:{location:?}, err:{source}",
```

File: wal/src/message_queue_impl/region.rs [hunk 4]
```diff
@@ -318,24 +327,32 @@ impl<M: MessageQueue> Region<M> {
...
+        region_safe_delete_offset: Option<Offset>,
         log_topic: &str,
         log_encoding: &CommonLogEncoding,
         builder: &mut RegionContextBuilder,
     ) -> Result<()> {
         info!(
-            "Recover region meta from log, namespace:{}, region id:{}, start offset:{}",
-            namespace, region_id, start_offset
+            "Recover region meta from log, namespace:{}, region_id:{}, region_safe_delete_offset:{:?}",
+            namespace, region_id, region_safe_delete_offset
```
Comment (ShiKaiWi) [inline wal/src/message_queue_impl/region.rs:344]: Doesn't the clippy complain about this?

Comment (Rachelint) [reply wal/src/message_queue_impl/region.rs]: No... clippy doesnt check it now.

**Review by ShiKaiWi — APPROVED** [review]
LGTM

**Pull request merged**